var/spack/repos/builtin/packages/mvapich-plus/package.py

```diff
@@ -0,0 +1,365 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import itertools
+import os.path
+import re
+import sys
+
+from spack.package import *
+
+
+class MvapichPlus(AutotoolsPackage):
+    """Mvapich is a High-Performance MPI Library for clusters with diverse
+    networks (InfiniBand, Omni-Path, Ethernet/iWARP, and RoCE) and computing
+    platforms (x86 (Intel and AMD), ARM and OpenPOWER)"""
+
+    homepage = "https://mvapich.cse.ohio-state.edu/userguide/userguide_spack/"
+    url = "https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich-3.0.tar.gz"
+    list_url = "https://mvapich.cse.ohio-state.edu/downloads/"
+
+    maintainers("natshineman", "harisubramoni", "MatthewLieber")
+
+    executables = ["^mpiname$", "^mpichversion$"]
+
+    license("Unlicense")
+
+    # Prefer the latest stable release
+    version("4.0", sha256="942156804425752ab8b7884a6995581d7d9e93f58025ca71b58e6412eb766eae")
+
+    provides("mpi")
+    provides("mpi@:3.1")
+
+    variant("wrapperrpath", default=True, description="Enable wrapper rpath")
+    variant("debug", default=False, description="Enable debug info and error messages at run-time")
+
+    variant("cuda", default=True, description="Enable CUDA extension")
+
+    variant("regcache", default=True, description="Enable memory registration cache")
```

Comment:
It's good practice to try and minimize the number of variants we default to `true`. For example, `cuda` on systems without nvidia hardware could cause issues / cause folks to have longer more complicated dependency trees without any benefit.